fairseq2n/src/fairseq2n/data/data_pipeline.cc

```diff
@@ -224,6 +225,39 @@ data_pipeline::round_robin(std::vector<data_pipeline> pipelines)
     return data_pipeline_builder{std::move(factory)};
 }
 
+data_pipeline_builder
+data_pipeline::sample(
+    std::vector<data_pipeline> pipelines,
+    std::optional<std::vector<float>> weights)
```

Comment:
Nit: could you please use the `float32` alias here to be consistent with the rest of the code. It is defined in [float.h](https://github.com/facebookresearch/fairseq2/blob/main/fairseq2n/src/fairseq2n/float.h).